src/Npgsql/NpgsqlActivitySource.cs

```diff
@@ -47,7 +47,7 @@ static class NpgsqlActivitySource
             throw new ArgumentOutOfRangeException(nameof(commandType), commandType, null);
         }
 
-        var activity = Source.StartActivity(activityName, ActivityKind.Client);
+        var activity = Source.StartActivity(string.IsNullOrEmpty(spanName) ? activityName : spanName, ActivityKind.Client);
```

Comment:
Activity can't have an empty name

![image](https://github.com/user-attachments/assets/293ecf14-7ef7-4948-9bdf-83ac95246aa5)

Reply:
I see, OK... Then I'd suggest just letting Activity throw if a user does that rather than treat it as null - if their expectation is to get an empty Activity it's better to get a clear exception than a silent unexpected behavior modification...